Assets/Scripts/TitleCameraMovement.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Moves an object on the title screen for the camera to follow
+/// </summary>
+public class TitleCameraMovement : MonoBehaviour
+{
+    [SerializeField]
+    [Tooltip("Object movement speed.")]
+    private float moveSpeed = 1;
+
+    [SerializeField]
+    [Tooltip("Maximum difference between X and Y velocity.")]
+    private float maximumDirectionalDifference = 0.75f;
+
+    [SerializeField]
+    [Tooltip("How far from the center does a collision need to be to change direction?")]
+    private float distanceMargin = 0.8f;
+
+    /// <summary>
+    /// The rigidbody attached to the object
+    /// </summary>
+    private Rigidbody2D rigidBody;
+
+    /// <summary>
+    /// The circle collider attached to the object
+    /// </summary>
+    private CircleCollider2D circleCollider;
+
+    /// <summary>
+    /// How far are the sides of the collider from its center?
+    /// </summary>
+    private float colliderRadius;
+
+    /// <summary>
+    /// Velocity of the object
+    /// </summary>
+    private Vector2 objectVelocity;
+    
+    // Start is called before the first frame update
+    void Start()
+    {
+        rigidBody = GetComponent<Rigidbody2D>();
+        circleCollider = GetComponent<CircleCollider2D>();
+        colliderRadius = circleCollider.radius * distanceMargin;
+        objectVelocity = GetRandomStartingVelocity() * moveSpeed;
+        rigidBody.velocity = objectVelocity;
+    }
+
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        Vector2 contactPoint = collision.GetContact(0).point;
```

Comment:
Try to make more of the intent of this logic clear. Put it in a named method with a summary, etc.